### PR TITLE
Added the German localization support and updated the language selection `text` to `flag emoji`

### PR DESCRIPTION
**Pull request by Hedayat-Farahi786**
**Hey @aykutkardas 👋**

**I've added the followings:**

* `German` localization support
* Changed the language selection `text` to `flag emojis`

Hope you like it!

Emojis Credit: https://emojipedia.org

### CODE REVIEW

File: src/localization/de/general.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  "general.prev": "Zurück",
+  "general.next": "Weiter",
+  "general.result": "Ergebnis",
+  "general.regex": "Regex",
+  "allgemein.text": "Text",
```
Comment (aykutkardas) [inline src/localization/de/general.json:6]: It's should be  `"general.text"`

File: src/components/LanguageSwitch/LanguageSwitch.module.css [hunk 2]
```diff
@@ -23,15 +23,17 @@
...
+  margin-left: 10px;
   color: var(--txt-1);
 }
 
 .LanguageSwitchList {
   position: absolute;
   top: 20px;
   z-index: 99;
-  right: 21px;
+  right: 0px;
```
Comment (aykutkardas) [inline src/components/LanguageSwitch/LanguageSwitch.module.css:36]: image

It's should be `25px`


image

File: src/components/LanguageSwitch/LanguageSwitch.module.css [hunk 3]
```diff
@@ -63,9 +65,13 @@
   display: inline-block;
 }
 
+.LanguageSwitchListItem:nth-child(1) {
+  margin-top: 10px;
+}
```
Comment (aykutkardas) [inline src/components/LanguageSwitch/LanguageSwitch.module.css:70]: This should be deleted. Instead, change the `top` value in the `.LanguageSwitchList` class to `30px`.

File: src/components/LanguageSwitch/LanguageSwitch.module.css (hunk 2 above)
Comment (Hedayat-Farahi786) [reply src/components/LanguageSwitch/LanguageSwitch.module.css:36]: Yeah, that looks great! 🚀

File: src/localization/de/general.json (hunk 1 above)
Comment (Hedayat-Farahi786) [reply src/localization/de/general.json:6]: Ohh, my bad 🤦‍♂️

Comment (Hedayat-Farahi786) [reply src/localization/de/general.json:6]: I might got confused 😑

File: src/localization/de/learn.json [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+  "steps.whatIsRegex.title": "Was sind reguläre Ausdrücke `Regex`?",
+  "steps.whatIsRegex.description": "Reguläre Ausdrücke sind eine Zeichenkette, die ein Suchmuster ausdrückt. Oft als `Regex` oder `Regexp` abgekürzt. Sie werden besonders verwendet, um Wörter in Texten zu finden oder zu ersetzen. Darüber hinaus können wir testen, ob ein Text den von uns festgelegten Regeln entspricht.\\n\\n Angenommen, Sie haben eine Liste mit Dateinamen und möchten nur Dateien mit der Erweiterung `pdf` finden. Geben Sie nach Eingabe eines Ausdrucks `^\\w+\\.pdf$` funktioniert. Die Bedeutung der Definitionen in diesem Ausdruck wird im Laufe der Schritte klarer.",
+
+  "steps.basicMatchers.title": "Grundlegende Matcher",
+  "steps.basicMatchers.description": "Das zu findende Zeichen oder Wort wird direkt geschrieben. Es ähnelt einem normalen Suchvorgang. Um beispielsweise das Wort `neugierig` im Text zu finden, geben Sie dasselbe ein.",
```
Comment (aykutkardas) [inline src/localization/de/learn.json:15]: `neugierig` This is the word intended to be in the text, so it should not be translated.

Expected: `curious`

File: src/localization/de/learn.json [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+  "examples.negatedCharacterSet.description": "Schreiben Sie den Ausdruck auf, der mit etwas anderem als den Wörtern `beor` und `beur` im Text übereinstimmt. Verwenden Sie dazu den negierten Zeichensatz.",
+
+  "examples.range.title": "Übung: Buchstabenbereich",
+  "examples.range.description": "Schreiben Sie den Ausdruck, der die Buchstaben von `g` bis `k` im Text auswählt.\\n Die Buchstaben `g` und `k` sollten ebenfalls in diesem Bereich enthalten sein.",
+
+  "examples.rangeNumber.title": "Übung: Nummernkreis",
+  "examples.rangeNumber.description": "Geben Sie einen Ausdruck ein, um Zahlen von `2` bis `7` im Text auszuwählen.\\n`2` und `7` sollten auch in diesem Bereich enthalten sein.",
+
+  "steps.repetitions.title": "Wiederholungen",
+  "steps.repetitions.description": "Einige Sonderzeichen werden verwendet, um anzugeben, wie oft ein Zeichen im Text wiederholt wird. Diese Sonderzeichen sind das Plus `+`, das Sternchen `*` und das Fragezeichen `? `.",
```
Comment (aykutkardas) [inline src/localization/de/learn.json:51]: **Current:** \`? `

**Expected:** \`?`

File: src/localization/de/learn.json [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+  "steps.nonCapturingGroupping.title": "Klammern `(?: )`: Gruppierung ohne Erfassung",
+  "steps.nonCapturingGroupping.description": "Sie können einen Ausdruck gruppieren und sicherstellen, dass er nicht durch Referenzen erfasst wird. Unten sehen Sie beispielsweise zwei Gruppen. Die erste Gruppenreferenz, die wir mit `\\1` bezeichnen, zeigt jedoch tatsächlich die zweite an Gruppe, da die erste eine nicht einfangende Gruppe ist.",
+
+  "steps.pipeCharacter.title": "Pipe-Zeichen `|`",
+  "steps.pipeCharacter.description": "Damit kann festgelegt werden, dass ein Ausdruck in verschiedenen Ausdrücken enthalten sein kann. Daher werden alle möglichen Anweisungen durch das Pipe-Zeichen `|` getrennt geschrieben. Dies unterscheidet sich vom Zeichensatz `[abc]`, Zeichensätze arbeiten auf Zeichenebene. Alternativen befinden sich auf Ausdrucksebene. Beispielsweise würde der folgende Ausdruck sowohl `Katze` als auch `Katze` auswählen. Fügen Sie am Ende des Ausdrucks ein weiteres Pipe-Zeichen `|` hinzu und geben Sie `Ratte` ein, damit alle Wörter werden ausgewählt.",
```
Comment (aykutkardas) [inline src/localization/de/learn.json:99]: This is the words intended to be in the text, so it should not be translated.
`katze`, `Katze`, `Ratte`

**Expected:** `cat`, `Cat`, `Rat`

File: src/localization/de/learn.json [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+  "steps.withoutNumberCharacter.description": "`\\D` wird verwendet, um nicht-numerische Zeichen zu finden.",
+
+  "steps.spaceCharacter.title": "Leerzeichen `\\s`",
+  "steps.spaceCharacter.description": "`\\s` wird verwendet, um nur Leerzeichen zu finden.",
+
+  "steps.withoutSpaceCharacter.title": "Außer Leerzeichen `\\S`",
+  "steps.withoutSpaceCharacter.description": "`\\S` wird verwendet, um Nicht-Leerzeichen zu finden.",
+
+  "steps.lookarounds.title": "Lookarounds",
+  "steps.lookarounds.description": "Wenn wir möchten, dass der Satz, den wir schreiben, vor oder nach einem anderen Satz steht, müssen wir \"umsehen\". Machen Sie den nächsten Schritt, um zu lernen, wie man \"umsieht\"." ,
```
Comment (aykutkardas) [inline src/localization/de/learn.json:129]: Use `lookarounds` instead of `umsehen`

File: src/localization/de/learn.json [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+  "steps.flagsMultiline.title": "Mehrzeiliges Flag",
+  "steps.flagsMultiline.description": "Regex sieht den gesamten Text als eine Zeile. Aber wir verwenden das `multiline`-Flag, um jede Zeile separat zu behandeln. Auf diese Weise schreiben wir die Ausdrücke entsprechend dem Zeilenende für jede Zeile separat . Aktivieren Sie nun das `multiline`-Flag, um alle Übereinstimmungen zu finden.",
+
+  "steps.flagsCaseInsensitive.title": "Groß-/Kleinschreibung beachten",
+  "steps.flagsCaseInsensitive.description": "Um die Groß-/Kleinschreibung des von uns geschriebenen Ausdrucks zu entfernen, müssen wir das Flag `Groß-/Kleinschreibung ignorieren` aktivieren.",
+  
+  "steps.greedyMatching.title": "Gieriger Abgleich",
+  "steps.greedyMatching.description": "Regex führt standardmäßig einen Greedy-Match durch. Das bedeutet, dass das Matchmaking so lang wie möglich sein wird. Sehen Sie sich das Beispiel unten an. Es bezieht sich auf alle Matches, die auf „r“ enden und beliebig sein können vorangestelltes Zeichen. Aber es hört nicht beim ersten Buchstaben `r` auf.",
```
Comment (aykutkardas) [inline src/localization/de/learn.json:156]: I'm not sure if it's a correct word.

`Matchmaking`

Comment (Hedayat-Farahi786) [reply src/localization/de/learn.json:156]: `Gieriger` == `Greedy` 
`Heiratsvermittler` == `Matchmaking`

Comment (Hedayat-Farahi786) [reply src/localization/de/learn.json:156]: Abgleich is kinda like a common saying word!

**Comment by aykutkardas** [comment]
> Yeah, your right! I made some typos even though I checked it all three times but still🤦‍♂️

I'm waiting for their fix. Then it's almost ready to merge. @Hedayat-Farahi786 ✨

File: src/components/LanguageSwitch/LanguageSwitch.module.css (hunk 2 above)
Comment (Hedayat-Farahi786) [reply src/components/LanguageSwitch/LanguageSwitch.module.css:36]: Or we can just use `left: 0px` instead of `right` value!

**Comment by Hedayat-Farahi786** [comment]
**Hey @aykutkardas,**

- Fixed all the typos
- Added fade-in-out animation to the languages list
- Added the scale transformation to the Flags on Hover

**Pull request merged**